samples/client/petstore/typescript-fetch/builds/oneOf/models/TestArrayResponse.ts

```diff
@@ -0,0 +1,89 @@
+/* tslint:disable */
+/* eslint-disable */
+/**
+ * testing oneOf without discriminator
+ * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
+ *
+ * The version of the OpenAPI document: 1.0.0
+ * 
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+import type { TestA } from './TestA';
+import {
+    instanceOfTestA,
+    TestAFromJSON,
+    TestAFromJSONTyped,
+    TestAToJSON,
+} from './TestA';
+import type { TestB } from './TestB';
+import {
+    instanceOfTestB,
+    TestBFromJSON,
+    TestBFromJSONTyped,
+    TestBToJSON,
+} from './TestB';
+
+/**
+ * @type TestArrayResponse
+ * 
+ * @export
+ */
+export type TestArrayResponse = Array<TestA> | Array<TestB> | Array<string>;
+
+export function TestArrayResponseFromJSON(json: any): TestArrayResponse {
+    return TestArrayResponseFromJSONTyped(json, false);
+}
+
+export function TestArrayResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): TestArrayResponse {
+    if (json == null) {
+        return json;
+    }
+    if (!(json instanceof Object)){
+        return {} as any;
+    }
+    if (Array.isArray(json) && json.every(item => item instanceof String)) {
```

Comment:
if these were [type guards](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates), Typescript can infer without the cast.
In this case for example:
```ts
function isStringArray(x: any): x is Array<string> {
    return Array.isArray(x) && x.every(item => item instanceof String);
}
```

see [playground](https://www.typescriptlang.org/play/?moduleResolution=99&target=7&module=6&resolvePackageJsonImports=true&allowArbitraryExtensions=false#code/FAMwrgdgxgLglgewgAjgZwMowE5wgcwEFtsBDATwAoAPALmVInIEp7rU1lizyAeNHHnwA+ZAG9gyKcmwBTGGGwpuFAHToVVas2QAyXcmqrZAN1nYqcGLIC2yALyirt1BAGMoshCGRZcBZgBuYABfYGAoJAFDB2QAbQAiUgSAGgSAIwSAXQZORnJg4DgQSnQ-IU0aZh0JaRl5RRQjG1IAB0pOR2Q0VUEbSmrQoA)

Reply:
Yes great idea, i'll implement it

Reply:
`item instanceof String` is incorrect. you need `typeof item === 'string'` instead.
you can check it in the console:
```
> "" instanceof String
< false
> typeof "" === 'string'
< true 
```

Reply:
and also modenr TS is smart enough to infer type guards, you don't need a custom type guard here.
here is a modified playground: https://www.typescriptlang.org/play/?moduleResolution=99&target=7&module=6&resolvePackageJsonImports=true&allowArbitraryExtensions=false#code/MYewdgzgLgBAHjAvDA2gIgIZoDRoEZoC6MGEJYAngNwBQNAlgGYAUAggE7sYUB09EHLhWZwAlDABkE+DwCmAN1nth9KLIC2SAHwwoFAA6yQjGKo1JEyAOTR29MAHMro8QG8aMT-A9eA9L5gAPQB+GgBfIA